operations/mimir/distributor.libsonnet

```diff
@@ -32,7 +27,21 @@
 
   distributor_ports:: $.util.defaultPorts,
 
-  distributor_env_map:: {},
+  distributor_env_map:: {
+    // Dynamically set GOMEMLIMIT based on memory request.
+    GOMEMLIMIT: std.toString(std.floor($.util.siToBytes($.distributor_container.resources.requests.memory))),
+
+    // We don't want to run GC unless GOMEMLIMIT is reached. We could use GOGC=off, but we use numeric value instead to
+    // run GC regularly even when GOMEMLIMIT is not reached yet.
```

Comment:
The two lines of the comment seem to contradict each other. I'm not sure what is meant here. I understand that setting GOGC=1000 will trigger a GC when the new heap size is 100x the size of the live heap size, and we will get more frequent GCs than with GOGC=off. Is there some minimum GC frequency or max % of GOMEMLIMIT we are targeting?